### DISCUSSION TITLE
ts-node with shebang vs ts-node-script

### DISCUSSION THREAD

**Discussion by ron23** [TypeStrong/ts-node#1186] [Q&A] [answered]
I posted this question https://github.com/TypeStrong/ts-node/issues/1007#issuecomment-751178630 and was suggested to ask it here. I didn't quite understand the response I got.
I went through the readme twice. The way I understand it:
* adding `ts-node-script` shebang enabled `--script-mode`.
* `ts-node-script scripts.ts` # Equivalent to ts-node --script-mode

So I thought that running `ts-node` on a script with shebang should use the local tsconfig file. But it doesn't.
As if the shebang doesn't matter.

### Example:
```
- foo
   - shebang.ts // with `ts-node-script`    
   - script.ts // with `node` shebang           
   - ts.config (module:commonJS)
- ts.config (module: esnext)
```
Both scripts uses `import`.

- `ts-node shebang.ts` // fails (uses the root ts.config)
- `ts-node node.ts` // fails - expected (uses the root ts.config)
- `ts-node -s shebang.ts` // works (uses the root ts.config)
- `ts-node -s node.ts` // works - expected (uses the root ts.config)
- `ts-node-script script.ts` //  works
- `ts-node-script shebang.ts`//  works

What am I missing?

**Comment by blakeembrey** [comment, marked as answer]
It sounds like some confusion around the shebang rather than `ts-node`. If I’m understanding the question correctly, you expected `ts-node x.ts` to automatically use the shebang? When running something with `ts-node` (or `node`) it doesn’t use the shebang anymore, it’s a feature of your Unix environment rather than the programming language.

To have your environment use the shebang, you’d invoke the script without node. For example, just do `./shebang.ts`.

**Comment by ron23** [comment]
Oh yea total misunderstanding of the shebang.
I get it now. Thanks!
